python/ray/monitor.py

```diff
@@ -127,38 +136,25 @@ def psubscribe(self, pattern):
         """
         self.primary_subscribe_client.psubscribe(pattern)
 
-    def xray_heartbeat_batch_handler(self, unused_channel, data):
-        """Handle an xray heartbeat batch message from Redis."""
-
-        pub_message = ray.gcs_utils.PubSubMessage.FromString(data)
-        heartbeat_data = pub_message.data
-
-        message = ray.gcs_utils.HeartbeatBatchTableData.FromString(
-            heartbeat_data)
-        for heartbeat_message in message.batch:
+    def get_all_heartbeat(self):
+        all_heartbeat = self.global_state_accessor.get_all_heartbeat()
+        for message in all_heartbeat:
+            heartbeat_message = ray.gcs_utils.HeartbeatTableData.FromString(
+                message)
             resource_load = dict(heartbeat_message.resource_load)
             total_resources = dict(heartbeat_message.resources_total)
             available_resources = dict(heartbeat_message.resources_available)
 
-            waiting_bundles, infeasible_bundles = \
-                parse_resource_demands(message.resource_load_by_shape)
-
-            pending_placement_groups = list(
-                message.placement_group_load.placement_group_data)
+            # TODO(wangtaothetonic): aggregate raylet load here
```

Comment:
Hmm this means when we merge this, it will break these right? Can we just update code here?

Reply:
ok will move p_g related fields into this interface.